Storage/ThirdParty/SwiftData.swift

```diff
@@ -422,12 +432,21 @@ public class SQLiteDBConnection {
 
     /// Closes a connection. This is called via deinit. Do not call this yourself.
     private func closeCustomConnection() -> NSError? {
-        let status = sqlite3_close(sqliteDB)
+        log.debug("Closing custom connection for \(self.filename) on \(NSThread.currentThread()).")
+        // Make sure we don't try to call sqlite3_close multiple times.
+        // TODO: add a lock here?
+        guard self.sqliteDB != nil else {
+            log.warning("Connection was nil.")
+            return nil
```

Comment:
For the record: `sqlite3_close` behaves correctly when the input is `nil`. I still think this is worth logging and returning.